rust/user-info-fetcher/src/http_error.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::Arc;
+
+use axum::{response::IntoResponse, Json};
+use hyper::StatusCode;
+use serde::Serialize;
+
+pub trait Error: std::error::Error {
```

Comment:
We might want to call that `ErrorExt`, as it extends the `std::error::Error` trait.

Reply:
Conventionally `FooExt` is for a trait adding new operations to `Ext`. A more appropriate name would be `HttpEndpointError`, but that's a mouthful.

Reply:
It does seem to be extending `std::error:Error` in the same way that the SubscriberExt trait extends Subscriber in the tracing crate.
https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.SubscriberExt.html

So maybe it should be ErrorExt

Reply:
`SubscriberExt` adds a *consumer-facing* convenience API on top of `Subscriber`'s *implementer-facing* API. You can see that [it has a single empty blanket impl](https://docs.rs/tracing-subscriber/0.3.18/src/tracing_subscriber/layer/mod.rs.html#1898).

The same pattern also holds elsewhere. [`futures::Future`](https://docs.rs/futures/0.3.29/futures/future/trait.Future.html) defines the base implementer-facing API (these days via `std`), [`FutureExt`](https://docs.rs/futures/0.3.29/futures/future/trait.FutureExt.html) defines the consumer-facing API as a blanket impl. [`TryFuture`](https://docs.rs/futures/0.3.29/futures/future/trait.TryFuture.html) defines a subset of `Future`s as fallible, [`TryFutureExt`](https://docs.rs/futures/0.3.29/futures/future/trait.TryFutureExt.html) provides the consumer-facing APIs that this implies. `TryFuture` itself is _not_ an `Ext` trait, because it carries its own semantics on top of the base trait.

`http_error::Error` is the opposite, it's a way for the error type to communicate _more_ information to the consumer, which isn't given by the std `Error`. There is no blanket impl, and there would be no way to provide a useful one.

Reply:
Thanks for the explanation. So this will stay as is (and will be moved as part of #493)